dali/base/dautils.hpp

```diff
@@ -418,4 +419,40 @@ interface ILocalOrDistributedFile: extends IInterface
 
 extern da_decl ILocalOrDistributedFile* createLocalOrDistributedFile(const char *fname,IUserDescriptor *user,bool onlylocal,bool onlydfs,bool iswrite=false);
 
+struct LockData
+{
+    unsigned mode;
+    SessionId sessId;
+    unsigned timeLockObtained;
+};
+
+typedef __int64 ConnectionId;
+typedef MapBetween<ConnectionId, ConnectionId, LockData, LockData> ConnectionInfoMap;
+
+enum LockDataField
+{
+    LDFXPath = 0,
+    LDFEPCount = 1,
+    LDFEP = 2,
+    LDFconnId = 3,
+    LDFsessId = 4,
+    LDFmode = 5,
+    LDFtimeLocked = 6
+};
```

Comment:
An exposed and probably extended LockData, e.g.
```
class CClientLockData : public LockData
```
that captures the extra info serialized should suffice, without extra IPT / enum layers.